sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -348,14 +348,18 @@ static int32_t _az_http_client_curl_upload_read_callback(
   if (userdata_length < 1)
     return CURLE_OK; // Success, all bytes copied
 
+  // Calculate how many bytes can we copy from customer data (upload_content)
+  // Curl provides dst buffer with a max size of dest_buffer_size, if customer data size is less
+  // than the max, we can copy all customer data directly and have it uploaded at once.
+  // If not, we can only copy the max dst size from customer data and wait for another upload to
+  // copy a next chunk of data
   int32_t size_of_copy = (userdata_length < dst_buffer_size) ? userdata_length : dst_buffer_size;
 
   memcpy(dst, az_span_ptr(*upload_content), (size_t)size_of_copy);
```

Comment:
consider using `az_span_copy` instead.

Reply:
We would need to make an az_span from dst, so it would be like adding an extra layer just to be able to use az_span.  Think it would be overkilling